test-utils/runtime/src/system.rs

```diff
@@ -192,6 +192,10 @@ pub fn validate_transaction(utx: Extrinsic) -> TransactionValidity {
 /// This doesn't attempt to validate anything regarding the block.
 pub fn execute_transaction(utx: Extrinsic) -> ApplyExtrinsicResult {
 	let extrinsic_index: u32 = storage::unhashed::get(well_known_keys::EXTRINSIC_INDEX).unwrap();
+	// Arbitrary limit on test-runtime block size.
+	if extrinsic_index > 15 {
```

Comment:
Could we not add a custom extrinsic type here: https://github.com/paritytech/substrate/blob/master/test-utils/runtime/src/lib.rs#L110

That always returns `ExhaustsResources` in `validate_transaction`?

Reply:
+1 on that, I remember several times it was hard figuring out what mocked result is returned and based on what

Reply:
Great idea. Addressed.